extensions/iceberg/src/main/java/io/deephaven/iceberg/util/IcebergUpdateMode.java

```diff
@@ -0,0 +1,57 @@
+//
+// Copyright (c) 2016-2024 Deephaven Data Labs and Patent Pending
+//
+package io.deephaven.iceberg.util;
+
+import io.deephaven.annotations.BuildableStyle;
+import org.immutables.value.Value;
+
+@Value.Immutable
+@BuildableStyle
+public abstract class IcebergUpdateMode {
```

Comment:
We may need another pass over this design. Would like input from @devinrsmith and @rbasralian .

Reply:
I think we might want to consider whether initial snapshot should be a parameter.